cmake/Dependencies.cmake

```diff
@@ -1,3 +1,14 @@
+IF (CMAKE_CXX_STANDARD GREATER_EQUAL 17)
+  SET(KOKKOS_SIMD_TEST_CLASS PT)
+ELSE()
+  SET(KOKKOS_SIMD_TEST_CLASS EX)
+  IF (${PROJECT_NAME}_ENABLE_KokkosSimd)
```

Comment:
Any reason why you preferred `${PROJECT_NAME}_` to `Kokkos_`?

Reply:
Because this is specifically for Trilinos and in Trilinos `${PROJECT_NAME}=Trilinos`, while in the standalone build `${PROJECT_NAME}=Kokkos`.

Reply:
I am not sure it is a good idea though, because it is at the end of the day a Kokkos option and in a world where Kokkos becomes a TPL for Trilinos it will surprise users.
Trilinos packages are free to introduce their own dependent option after the fact if they like to do so.

Reply:
It is not just a good idea, it is the only correct solution. We are not in a world where Kokkos is a TPL for Trilinos. Kokkos is a Trilinos package that gets snapshotted into Trilinos. When that happens this file decides whether the `KokkosSimd` sub-package of Trilinos is disabled by default or not, and Trilinos users *have to* use `Trilinos_ENABLE_KokkosSimd` to enable it otherwise, `Kokkos_ENABLE_KokkosSimd` does nothing. I am not introducing anything new or confusing here, I am working with the TriBITS build that Kokkos has to support for now.

Reply:
This is not a Kokkos option. It is only an option in the TriBITS/Trilinos build.